### PR TITLE
[LoRA] feat: support non-diffusers lumina2 LoRAs.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Fixes:
https://github.com/huggingface/diffusers/issues/10866

Cc: @nitinmukesh

Results (under the same seed)

| LoRA | No LoRA |
|---------|---------|
| image | image |

<details>
<summary>Code</summary>

```py
import torch
from diffusers import Lumina2Text2ImgPipeline
from peft.tuners.tuners_utils import BaseTunerLayer

pipe = Lumina2Text2ImgPipeline.from_pretrained(
    "Alpha-VLLM/Lumina-Image-2.0", torch_dtype=torch.bfloat16
)

# Art Style of Hitoshi Ashinano https://civitai.com/models/1269546/art-style-of-hitoshi-ashinano-lumina-image-20
pipe.load_lora_weights("newgenai79/lumina2_lora",weight_name="Art_Style_of_Hitoshi_Ashinano.safetensors")
pipe.enable_model_cpu_offload()

prompt = "Hitoshi Ashinano style A young girl with vibrant green hair and large purple eyes peeks out from behind a white wooden door. She is wearing a white shirt and have a curious expression on her face. The background shows a blue sky with a few clouds, and there's a white fence visible. Green leaves hang down from the top left corner, and a small white circle can be seen in the sky. The scene captures a moment of innocent curiosity and wonder."
image = pipe(
    prompt, 
    negative_prompt="blurry, ugly, bad, deformed iris, deformed pupils, semi-realistic, cgi, 3d, render, cropped, out of frame, worst quality, low quality, jpeg artifacts, fused fingers, morbid, mutilated, extra fingers, mutated hands, bad anatomy, bad proportion, extra limbs", 
    guidance_scale=6,
    num_inference_steps=35, 
    generator=torch.manual_seed(0)
).images[0]
image.save("lumina2_lora.png")

```

</details>

### CODE REVIEW

**Review by hlky — APPROVED** [review]

**Review by shethaadit — APPROVED** [review]

**Comment by nitinmukesh** [comment]
Awesome, thank you @sayakpaul .

Looking forward to getting it merged.

**Comment by sayakpaul** [comment]
@DN6 the failing tests seem to be unrelated to this PR?

File: src/diffusers/loaders/lora_pipeline.py [hunk 1]
```diff
@@ -3909,6 +3909,11 @@ def lora_state_dict(
             logger.warning(warn_msg)
             state_dict = {k: v for k, v in state_dict.items() if "dora_scale" not in k}
 
+        # conversion.
+        non_diffusers = any(k.startswith("diffusion_model.") for k in state_dict)
+        if non_diffusers:
+            state_dict = _convert_non_diffusers_lumina2_lora_to_diffusers(state_dict)
```
Comment (DN6) [inline src/diffusers/loaders/lora_pipeline.py:3915]: Is this prefix specific to Lumina? Should we always just remove it?

Comment (sayakpaul) [reply src/diffusers/loaders/lora_pipeline.py:3915]: It is not specific to Lumina2 but specific to external trainer libraries. In all the past iterations where we have supported non-diffusers LoRA checkpoints, we have removed it because it's not in the `diffusers` compatible format. 

So, I am confident that we should always remove it.

**Review by DN6 — APPROVED** [review]
